src/analysis.rs

```diff
@@ -57,6 +83,22 @@ impl Cost {
     pub const fn from_milliweight(milliweight: usize) -> Self {
         Cost(milliweight)
     }
+
+    /// Return whether the cost is allowed by consensus.
+    ///
+    /// This means the cost is within the maximum budget
+    /// that any program inside a Taproot transaction can have.
+    pub fn is_consensus_valid(&self) -> bool {
+        self <= &Self::CONSENSUS_MAX
+    }
+
+    /// Return whether the cost is within the budget of
+    /// the given transaction input.
+    #[cfg(feature = "elements")]
+    pub fn is_budget_valid(&self, txin: &elements::TxIn) -> bool {
+        let budget = txin.witness.script_witness.iter().map(|el| el.len()).sum();
+        self.to_weight() <= budget
```

Comment:
In 00683984959400de3dd8c4902bc4940c5ded159a:

You should replace this with `txin.witness.serialized_len()` + 50_000 to match the consensus code at https://github.com/ElementsProject/elements/blob/simplicity/src/script/interpreter.cpp#L3301

Reply:
Good find. I wrote about the validation offset in the documentation, but I forgot it in this method.

Reply:
I pushed an update.